charts/terraform-pod/templates/deployment.yaml

```diff
@@ -0,0 +1,165 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "terraform-pod.fullname" . }}
+  labels:
+    {{- include "terraform-pod.labels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "terraform-pod.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
```

Comment:
Let's reorganize `annotations` and `imagePullSecrets` blocks to look the same as `labels` and `securityContext`, i.e.
```
     imagePullSecrets:
          {{- toYaml .Values.imagePullSecrets | nindent 8 }}
```

so everything would look the same (there would be no different formatting for similar things)

Reply:
`with` in the way it is written replaces (if) check as these attrs are optional. That is the default state of the code when you create the chart using `helm create`

```yaml
{{- with .Values.imagePullSecrets }}
imagePullSecrets:
  {{- toYaml . | nindent 8 }}
{{- end }}
```

So I keep it the way it is.

Reply:
Ok, I didn't know that it replaces if

Reply:
@kseniyashaydurova yes it does.